backend/census_historical_migration/test_federal_awards_xforms.py

```diff
@@ -588,6 +589,41 @@ def test_match_numbers_mixed_empty_and_non_empty(self):
         self.assertEqual(transformed_ids, expected_ids)
 
 
+class TestXformMissingMajorProgram(SimpleTestCase):
+    class AuditMock:
+        def __init__(
+            self,
+            major_program,
+            audit_type,
+        ):
+            self.MAJORPROGRAM = major_program
+            self.TYPEREPORT_MP = audit_type
+
+    def test_xform_normal_major_program(self):
+        """Test for normal major program"""
+        audits = [self.AuditMock("Y", "U")]
+
+        xform_missing_major_program(audits)
+
+        self.assertEqual(audits[0].MAJORPROGRAM, "Y")
+
+    def test_xform_missing_major_program_with_audit_type(self):
+        """Test for missing major program with audit type provided"""
+        audits = [self.AuditMock("", "U")]
+
+        xform_missing_major_program(audits)
+
+        self.assertEqual(audits[0].MAJORPROGRAM, "Y")
+
+    def test_xform_missing_major_program_without_audit_type(self):
+        """Test for missing major program without audit type provided"""
+        audits = [self.AuditMock("", "")]
+
+        xform_missing_major_program(audits)
+
+        self.assertEqual(audits[0].MAJORPROGRAM, "N")

```

Comment:
Shouldn't the result of the transformation be checked here?

Reply:
See other comment